File: android/src/main/kotlin/co/quis/flutter_contacts/AccountInfo.kt
package co.quis.flutter_contacts

data class AccountInfo(
    var name: String,
    var type: String,
    var accountId: String,
    var contactCount: Int,
    var addressCount: Int,

) {
    companion object {
        fun fromMap(m: Map<String, Any?>): AccountInfo {
            return AccountInfo(
                m["name"] as String,
                m["type"] as String,
                m["accountId"] as String,
                m["contactCount"] as Int,
                m["addressCount"] as Int,
            )
        }
    }

    fun toMap(): Map<String, Any?> = mapOf(
        "name" to name,
        "type" to type,
        "accountId" to accountId,
        "contactCount" to contactCount,
        "addressCount" to addressCount,

    )
}

File: ios/Classes/AccountInfo.swift
import Contacts

@available(iOS 9.0, *)
struct AccountInfo {
    var rawId: String
    // local, exchange, cardDAV or unassigned
    var type: String
    var name: String
    var accountId: String

    var contactCount: Int
    var addressCount: Int

    init(fromMap m: [String: Any?]) {
        rawId = m["rawId"] as! String
        type = m["type"] as! String
        name = m["name"] as! String
        accountId = m["accountId"] as! String
        contactCount = m["contactCount"] as! Int
        addressCount = m["addressCount"] as! Int
    }

    func toMap() -> [String: Any?] { [
        "rawId": rawId,
        "type": type,
        "name": name,
        "accountId": accountId,
        "contactCount": contactCount,
        "addressCount": addressCount,
    ]
    }
}
